bam_fastq.c

```diff
@@ -580,6 +592,7 @@ static inline bool filter_it_out(const bam1_t *b, const bam2fq_state_t *state)
 {
     return ((b->core.flag&(state->flag_on)) != state->flag_on // or reads indicated by filter flags
         ||  (b->core.flag&(state->flag_off)) != 0
+        ||  (((b->core.flag&(state->flag_anyon)) == 0) && (state->flag_anyon != 0))
```

Comment:
`--rf` would filter out reads if none of the bits match, given that there is a nonzero value for `state->flag_anyon`.